common/remote/sockfile.cpp

```diff
@@ -2583,11 +2583,28 @@ void CRemoteFile::copyTo(IFile *dest, size32_t buffersize, ICopyFileProgress *pr
 }
 
 
-unsigned getRemoteVersion(ISocket * socket, StringBuffer &ver)
+unsigned getRemoteVersion(ISocket * _socket, StringBuffer &ver)
 {
+    Linked<ISocket> socket = _socket;
```

Comment:
I don't think you need/want this.
The socket was passed in before without taking ownership.
You conditionally create a new (secure) socket below, which will need to be owned in this scope and you'll want to link the original (passed in) socket when you give it to createSecureSocket.
But otherwise the original doesn't need touching.